File: Java-Source/Day28.java
//DAY 28 OF HACKERRANK 30 DAY CHALLENGE: REGEX, PATTERNS, AND INTRO TO DATABASES (JAVA)

import java.util.*;

public class Day28 {

    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);
        int n = scan.nextInt();
        String[] inputs = new String[n];
        String[] names = new String[n];
        ArrayList<String> outputs = new ArrayList<String>();

        scan.nextLine();
        for (int x = 0; x < n; x++) {
            inputs[x] = scan.nextLine();
            String[] temp = inputs[x].split(" ");
            names[x] = temp[0];

            if (inputs[x].contains("@gmail.com")) {
                outputs.add(names[x]);
            }
        }

        for (int y = 0; y < outputs.size()-1; y++) {
            int minIndex = y;
            for (int z = y+1; z < outputs.size(); z++) {
                if (outputs.get(minIndex).compareTo(outputs.get(z)) > 0) {
                    minIndex = z;
                }
            }
            if (minIndex != y) {
                String temp = outputs.get(y);
                outputs.set(y,outputs.get(minIndex));
                outputs.set(minIndex,temp);
            }
        }

        for (int i = 0; i < outputs.size(); i++) {
            System.out.println(outputs.get(i));
        }
    }
}

File: CPP-Source/Day28.cpp
//DAY 28 OF HACKERRANK 30 DAY CHALLENGE: REGEX, PATTERNS, AND INTRO TO DATABASES (C++)

#include <iostream>
#include <string>
#include <vector>

using namespace std;


int main() {
	int N;
	cin >> N;
	vector<string> validIds = vector<string>(0);

	for (int a0 = 0; a0 < N; a0++) {
		string firstName;
		string emailID;
		cin >> firstName >> emailID;

		if (emailID.find("@gmail.com") != string::npos) {
			vector<string> temp = vector<string>(validIds.size() + 1);
			for (unsigned int x = 0; x < validIds.size(); x++) {
				temp[x] = validIds[x];
			}
			temp[validIds.size()] = firstName;
			validIds = temp;
		}
	}

	for (unsigned s = 0; s < validIds.size() - 1; s++) {
		int minIndex = s;
		for (int t = s + 1; t < validIds.size(); t++) {
			if (validIds[t].compare(validIds[minIndex]) < 0) {
				minIndex = t;
			}
		}
		if (minIndex != s) {
			string temp = validIds[s];
			validIds[s] = validIds[minIndex];
			validIds[minIndex] = temp;
		}
	}

	for (unsigned int z = 0; z < validIds.size(); z++) {
		cout << validIds[z] << endl;
	}

	return 0;
}
